### PR TITLE
CB-19712 Extend the response with more information

### PR DESCRIPTION
**Pull request by votez**
(cherry picked from commit e2e2c1bfa77791ee95a92e0a3dcd650a96d03707)

CB-20036 Fix SdxBackupStatusResponse deserialization

Added fields :
 1.  _timestamp_ for the last backup date (start date of the backup)
 1. _includedData_ for the backup contents (collection). Possible values: 
      * RANGER_AUDITS
      * RANGER_PERMISSIONS
      * HMS_METADATA
      * ATLAS_METADATA
      * ATLAS_INDEXES

Implementation notes:
* DatalakeRestoreStatusResponse does not extends DatalakeBackupStatusResponse anymore since these are logically different operations. Instead, they both are now extend DatalakeOperationStatus
* Moved isComplete and isFailed to Status enum since it belongs to Status but not Response (left delegate methods for backward compatibility)
* refactoring included : replaced java _assert_ with JUnit _assertXXX_ in tests 

Sample output:
```json
{
    "operationId": "61afae5b-6ab0-4fc4-a8de-9cc30aaf7337",
    "status": "SUCCESSFUL",
    "reason": "No failure messages found",
    "includedData": [
        "RANGER_AUDITS",
        "RANGER_PERMISSIONS",
        "HMS_METADATA",
        "ATLAS_METADATA",
        "ATLAS_INDEXES"
    ],
    "timestamp": "2022-11-22T14:59:56.000895+00:00"
```

tested as:
1) created backup via cdp-cli
1) verified new fields are presented in HTTP request for backup status
1) restored backup via cdp-cli
1) verified status on CDP UI

### CODE REVIEW

**Comment by lnardai** [comment]
Agreed with @kkalvagadda1 , can you retarget this to master? then we can merge it.

**Comment by votez** [comment]
> Agreed with @kkalvagadda1 , can you retarget this to master? then we can merge it.

@lnardai, @kkalvagadda1  retargeted the PR

**Review by kkalvagadda1 — APPROVED** [review]

**Review by lnardai — APPROVED** [review]

File: datalake-dr-connector/src/main/java/com/sequenceiq/cloudbreak/datalakedr/converter/GrpcStatusResponseToDatalakeBackupRestoreStatusResponseConverter.java [hunk 1]
```diff
@@ -29,36 +30,58 @@ public class GrpcStatusResponseToDatalakeBackupRestoreStatusResponseConverter {
...
         );
     }
 
     public DatalakeBackupStatusResponse convert(BackupDatalakeStatusResponse response) {
+        List<String> contents = new LinkedList<>();
+        InternalBackupRestoreState operationStates = response.getOperationStates();
+        SolrBackupRestoreState solrState = operationStates.getSolr();
+        HbaseBackupRestoreState hbaseState = operationStates.getHbase();
+
+        if (solrState.getRangerAuditsCollection().getStatus().equals(SUCCESSFUL_STATUS)) {
```
Comment (doktoric) [inline datalake-dr-connector/src/main/java/com/sequenceiq/cloudbreak/datalakedr/converter/GrpcStatusResponseToDatalakeBackupRestoreStatusResponseConverter.java:58]: dontt we need to handle any null here?

Comment (votez) [reply datalake-dr-connector/src/main/java/com/sequenceiq/cloudbreak/datalakedr/converter/GrpcStatusResponseToDatalakeBackupRestoreStatusResponseConverter.java:58]: these (all the chain of objects) are from Protobuf generated definitions - they are all non-null

**Pull request merged**